network/tests/fuzzing.rs

```diff
@@ -137,3 +141,425 @@ async fn fuzzing_post_handshake() {
+            let corrupted_payload: Vec<u8> = serialized
+                .into_iter()
+                .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
+                .collect();
+
+            let header = MessageHeader::from(corrupted_payload.len());
+
+            node1.write_bytes(&header.as_bytes()).await;
+            node1.write_bytes(&corrupted_payload).await;
+        }
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_payloads_with_hashes_pre_handshake() {
+    // tracing_subscriber::fmt::init();
+
+    let node_setup = TestSetup {
+        consensus_setup: None,
+        ..Default::default()
+    };
+
+    let node = test_node(node_setup).await;
+    let node_addr = node.local_address().unwrap();
+
+    let mut rng = thread_rng();
+
+    let hashes: Vec<BlockHeaderHash> = (0u8..10).map(|i| BlockHeaderHash::new(vec![i; 32])).collect();
+
+    for payload in &[
+        Payload::GetBlocks(hashes.clone()),
+        Payload::GetSync(hashes.clone()),
+        Payload::Sync(hashes),
+    ] {
+        for _ in 0..ITERATIONS {
+            let serialized = Payload::serialize(payload).unwrap();
+            let corrupted_payload: Vec<u8> = serialized
+                .into_iter()
+                .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
+                .collect();
+
+            let header = MessageHeader::from(corrupted_payload.len());
+
+            let mut stream = TcpStream::connect(node_addr).await.unwrap();
+            let _ = stream.write_all(&header.as_bytes()).await;
+            let _ = stream.write_all(&corrupted_payload).await;
+        }
+    }
+
+    assert_eq!(node.peer_book.read().number_of_connected_peers(), 0);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_payloads_with_hashes_post_handshake() {
+    // tracing_subscriber::fmt::init();
+
+    let (node1, mut node2) = spawn_2_fake_nodes().await;
+
+    tokio::spawn(async move {
+        loop {
+            let _ = node2.read_payload().await;
+        }
+    });
+
+    let mut rng = thread_rng();
+
+    let hashes: Vec<BlockHeaderHash> = (0u8..10).map(|i| BlockHeaderHash::new(vec![i; 32])).collect();
+
+    for payload in &[
+        Payload::GetBlocks(hashes.clone()),
+        Payload::GetSync(hashes.clone()),
+        Payload::Sync(hashes),
+    ] {
+        for _ in 0..ITERATIONS {
+            let serialized = Payload::serialize(payload).unwrap();
+            let corrupted_payload: Vec<u8> = serialized
+                .into_iter()
+                .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
+                .collect();
+
+            let header = MessageHeader::from(corrupted_payload.len());
+
+            node1.write_bytes(&header.as_bytes()).await;
+            node1.write_bytes(&corrupted_payload).await;
+        }
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_currupted_peers_pre_handshake() {
```

Comment:
couldn't `ping` and `peers` be joined with the `payloads_with_blobs` tests? they could be renamed `payloads_with_bodies` so that it still work, and it would deduplicate the code